
From: Geert Uytterhoeven <geert@linux-m68k.org>

I/O port numbers can be larger than 8-bit on many platforms (this caused a
warning when {out,in}b() cast reg to a pointer on platforms with memory
mapped I/O)


---

 25-akpm/drivers/ide/legacy/qd65xx.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff -puN drivers/ide/legacy/qd65xx.c~QD65xx-io-ports-fix drivers/ide/legacy/qd65xx.c
--- 25/drivers/ide/legacy/qd65xx.c~QD65xx-io-ports-fix	Tue Apr  6 15:33:31 2004
+++ 25-akpm/drivers/ide/legacy/qd65xx.c	Tue Apr  6 15:33:31 2004
@@ -92,7 +92,7 @@
 
 static int timings[4]={-1,-1,-1,-1}; /* stores current timing for each timer */
 
-static void qd_write_reg (u8 content, u8 reg)
+static void qd_write_reg (u8 content, unsigned long reg)
 {
 	unsigned long flags;
 
@@ -101,7 +101,7 @@ static void qd_write_reg (u8 content, u8
 	spin_unlock_irqrestore(&ide_lock, flags);
 }
 
-u8 __init qd_read_reg (u8 reg)
+u8 __init qd_read_reg (unsigned long reg)
 {
 	unsigned long flags;
 	u8 read;

_
